File: Python/uitests/pages/main_page.py
class MainPage():
    def __init__(self, driver):
        self.driver = driver
        self.driver.implicitly_wait(100)
        self._search_bar = None
        self._main_title = None
        self._expected_result = None

    @property
    def search_bar(self):
        from elements.search_bar import SearchBar

        if self._search_bar is None:
            self._search_bar = SearchBar(self.driver, self.driver.find_element_by_xpath("//*[contains(@class, 'searchbar')]"))
        return self._search_bar

    @property
    def main_title(self):
        from elements.main_title import MainTitle

        if self._main_title is None:
            self._main_title = MainTitle(self.driver, self.driver.find_element_by_xpath("//*[contains(@class, 'text-title')]"))
        return self._main_title

    @property
    def page_source(self):
        return self.driver.page_source

    # @property
    # def expected_result(self):
    #   from expected_result import ExpectedResult
    #   if self._expected_result is None:
    #      self._expected_result = ExpectedResult(self.driver,

    #                                        self.driver.find_elements_by_class_name(ExpectedResult.selectors['expected_text']))
    # return self._expected_result

    def open(self, account=None):
        url = "http://workspace19.test.crm.2gis.ru"
        if account is not None:
            url = url + "?me=" + account
        self.driver.get(url)

    @property
    def title(self):
        return self.driver.title

File: C#/TestProject2(PageObj)/UnitTestProject1/Pages/MainPage.cs
using System.Collections.Generic;
using System.Linq;

using DoubleGis.Erm.UnitTestProject1.Elements;

using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace DoubleGis.Erm.UnitTestProject1.Pages
{
   public class MainPage
   {
       private string baseUrl = "http://workspace19.test.crm.2gis.ru";
       public IWebDriver WebDriver { get; set; }
       private IWebElement mainTittle;

        public MainPage(IWebDriver webDriver)
        {
            WebDriver = webDriver;
            PageFactory.InitElements(webDriver, this);
        }

       [FindsBy(How = How.XPath, Using = "//*[contains(@class,'title__text')]")]
       public IWebElement MainTitle { get; set; } 

      public SearchBar SearchBar => new SearchBar(WebDriver.FindElement(By.XPath("//*[contains(@class, 'searchbar')]")));

       public IList<UserDeal> Deals => GetUserDealolletion();

       public IWebDriver Open(string account = null)
       {
           if (account != null)
           {
               baseUrl = $"{baseUrl}?me={account}";
           }

           WebDriver.Navigate().GoToUrl(url: baseUrl);
           return WebDriver;
       }

       private IList<UserDeal> GetUserDealolletion()
       {
           var userdealList = WebDriver.FindElements(By.ClassName("deal"));

           return userdealList.Select(x => new UserDeal(x)).ToList();
       }
   }
}
